R/match_pc_gpprac.R

```diff
@@ -0,0 +1,256 @@
+#' Match postcode with GP Practice
+#'
+#' @description Match postcode with GP Practice
+#'
+#' @param data episode files
+#'
+#' @return data with matched postcode
+#' @export
+match_pc_gpprac <- function(data) {
+  # Recoding hb codes to 2018 standard ----
+  data_hb_pc <- data %>%
+    dplyr::mutate(hbtreatcode = as.character.Date(hbtreatcode)) %>%
+    dplyr::mutate(
+      dplyr::across(
+        c("hbrescode", "hbpraccode", "hbtreatcode"),
+        ~ dplyr::case_when(
+          # HB2014 to HB2018
+          .data == "S08000018" ~ "S08000029",
+          .data == "S08000027" ~ "S08000030",
+          # HB2019 to HB2018
+          .data == "S08000031" ~ "S08000021",
+          .data == "S08000032" ~ "S08000023",
+          TRUE ~ .
+        )
+      ),
+      HSCP = dplyr::case_when(
+        # HSCP2016 to HSCP2018
+        HSCP == "S37000014" ~ "S37000032",
+        HSCP == "S37000023" ~ "S37000033",
+        # HSCP2019 to HSCP2018
+        HSCP == "S37000034" ~ "S37000015",
+        HSCP == "S37000035" ~ "S37000021",
+        TRUE ~ HSCP
+      )
+    ) %>%
+    # Making postcodes into 7-character format ----
+    dplyr::mutate(
+      postcode = phsmethods::format_postcode(postcode, format = "pc7")
+    )
+
+  ## Rename to keep the existing geographies for now, in case the postcode can't be matched ----
+  data_hb_pc_1 <- data_hb_pc %>%
+    dplyr::rename(
+      lca_old = "lca",
+      hscp_old = "HSCP",
+      datazone_old = "DataZone",
+      hbrescode_old = "hbrescode"
+    )
+
+  pc_lookup <-
+    readr::read_rds(fs::path(
+      get_slf_dir(),
+      "Lookups/source_postcode_lookup_Sep_2022.rds"
+    ))
```

Comment:
I think this path already exists as a function

Reply:
get_slf_postcode_path()

Reply:
Changed to use this function